src/nfa/delta.cc

```diff
@@ -374,6 +375,24 @@ bool Delta::operator==(const Delta& other) const {
     return other_transitions_it == other_transitions_end;
 }
 
+//Returns an iterator to the smallest epsilon, or end() if there is no epsilon
+//Searches from the end of the vector of SymbolPosts, since epsilons are at the end and they are typically few, mostly 1.
+StatePost::const_iterator StatePost::first_epsilon_it(Symbol first_epsilon) const {
+    auto it = end();
```

Comment:
I am doing this to avoid duscussions,
but to be honest, it is slightly pissing me off, I it is a premature optimization that makes code clumsy and has no impact on speed